src/libraries/System.Private.CoreLib/src/System/Boolean.cs

```diff
@@ -236,8 +236,14 @@ public static bool Parse(string value)
             return Parse(value.AsSpan());
         }
 
-        public static bool Parse(ReadOnlySpan<char> value) =>
-            TryParse(value, out bool result) ? result : throw new FormatException(SR.Format(SR.Format_BadBoolean, new string(value)));
+        public static bool Parse(ReadOnlySpan<char> value)
+        {
+            if (!TryParse(value, out bool result))
+            {
+                ThrowHelper.ThrowFormatException_BadBoolean(value);
```

Comment:
Replaced throw with a throw helper to make it inlineable (at least for constant input)